cljfmt/src/cljfmt/main.clj

```diff
@@ -43,16 +43,28 @@
 (defn- project-path [{:keys [project-root]} file]
   (-> project-root (or ".") io/file (relative-path (io/file file))))
 
-(defn- format-diff
-  ([options file]
-   (let [original (slurp (io/file file))]
-     (format-diff options file original (reformat-string options original))))
-  ([options file original revised]
-   (let [filename (project-path options file)
-         diff     (diff/unified-diff filename original revised)]
-     (if (:ansi? options)
-       (diff/colorize-diff diff)
-       diff))))
+(defmacro if-bb [then else]
+  (if (System/getProperty "babashka.version")
+    then else))
```

Comment:
Can you explain why this is a macro, instead of something like:

```clojure
(def bb? (boolean (System/getProperty "babashka.version")))
```

Reply:
We can do that if you prefer, but imo it's cleaner to not have any side effects of macro-expansions like `defn` happening at all, which is what a macro like that prevents.

Reply:
Does Babashka have a reader conditional in that case? i.e. would `#?(:bb ...)` work?

Reply:
@weavejester I solved this using a more elegant solution: I added a `.bb` file (which is always loaded first in bb) with the same namespace and vars so the `main` namespace doesn't need any changes. A reader conditional would have worked, but then we'd needed to rename the `.clj` namespace to `.cljc`.